File: api/golang/ToolGood/TextFilter/Api/Impl/TextFilterAsyncProvider.go
package Impl

import "encoding/json"
import "net/http"
import "net/url"
import	. "../Datas/Requests"
import	"io/ioutil"
import	"strconv"
import	. "../Datas/Texts"


type TextFilterAsyncProvider struct{
	textFilterHost string
}

func NewTextFilterAsyncProvider(config string) *TextFilterAsyncProvider{
	provider := &TextFilterAsyncProvider{}
	provider.textFilterHost = config
	return  provider;
}


const textFilterUrl_async string = "/api/async/text-filter";
const textReplaceUrl_async string = "/api/async/text-replace";
const htmlFilterUrl_async string = "/api/async/html-filter";
const htmlReplaceUrl_async string = "/api/async/html-replace";
const jsonFilterUrl_async string = "/api/async/json-filter";
const jsonReplaceUrl_async string = "/api/async/json-replace";
const markdownFilterUrl_async string = "/api/async/markdown-filter";
const markdownReplaceUrl_async string = "/api/async/markdown-replace";


 

func (this *TextFilterAsyncProvider)TextFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+textFilterUrl_async;
	return this.textFilter(request,u);
}
func (this *TextFilterAsyncProvider)HtmlFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+htmlFilterUrl_async;
	return this.textFilter(request,u);
}
func (this *TextFilterAsyncProvider)JsonFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+jsonFilterUrl_async;
	return this.textFilter(request,u);
}
func (this *TextFilterAsyncProvider)MarkdownFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+markdownFilterUrl_async;
	return this.textFilter(request,u);
}


func (this *TextFilterAsyncProvider)TextReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+textReplaceUrl_async;
	return this.textReplace(request,u);
}
func (this *TextFilterAsyncProvider)HtmlReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+htmlReplaceUrl_async;
	return this.textReplace(request,u);
}
func (this *TextFilterAsyncProvider)JsonReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+jsonReplaceUrl_async;
	return this.textReplace(request,u);
}
func (this *TextFilterAsyncProvider)MarkdownReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+markdownReplaceUrl_async;
	return this.textReplace(request,u);
}



func (this *TextFilterAsyncProvider)textReplace(request *TextReplaceRequest,u string)*TextReplaceResult   {
	resp, err := http.PostForm(u, url.Values{ "txt":{request.Txt}, "replaceChar":{request.ReplaceChar}, "reviewReplace":{strconv.FormatBool(request.ReviewReplace)}, "contactReplace":{strconv.FormatBool(request.ContactReplace)}, "skipBidi":{strconv.FormatBool(request.SkipBidi)}, "onlyPosition":{strconv.FormatBool(request.OnlyPosition)} })
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *TextReplaceResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}


func (this *TextFilterAsyncProvider)textFilter(request *TextFilterRequest,u string)*TextFilterResult   {
	resp, err := http.PostForm(u, url.Values{ "txt":{request.Txt}, "skipBidi":{strconv.FormatBool(request.SkipBidi)}, "onlyPosition":{strconv.FormatBool(request.OnlyPosition)}	}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *TextFilterResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}

File: api/python/Impl/TextFilterAsyncProvider.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# TextFilterAsyncProvider.py
# 2021, Lin Zhijun, https://github.com/toolgood/ToolGood.TextFilter.Api
# MIT Licensed 
 
__all__ = ['TextFilterAsyncProvider']
import json
import http.client, urllib.parse
import requests


class TextFilterAsyncProvider():

    def __init__(self,textFilterConfig):
        self.__textFilterConfig=textFilterConfig
        self.__TextFilterUrl = "/api/async/text-filter"
        self.__TextReplaceUrl = "/api/async/text-replace"
        self.__HtmlFilterUrl = "/api/async/html-filter"
        self.__HtmlReplaceUrl = "/api/async/html-replace"

        self.__JsonFilterUrl = "/api/async/json-filter"
        self.__JsonReplaceUrl = "/api/async/json-replace"
        self.__MarkdownFilterUrl = "/api/async/markdown-filter"
        self.__MarkdownReplaceUrl = "/api/async/markdown-replace"
    
    def TextFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig, self.__TextFilterUrl,request)
        return json.loads(result)

    def TextReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig,self.__TextReplaceUrl,request)
        return json.loads(result)

    def HtmlFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        
        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig,self.__HtmlFilterUrl,request)
        return json.loads(result)

    def HtmlReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig,self.__HtmlReplaceUrl,request)
        return json.loads(result)

    def JsonFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        
        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig,self.__JsonFilterUrl,request)
        return json.loads(result)

    def JsonReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        
        result=doPost(self.__textFilterConfig,self.__JsonReplaceUrl,request)
        return json.loads(result)
 
    def MarkdownFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        
        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig,self.__MarkdownFilterUrl,request)
        return json.loads(result)

    def MarkdownReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        @requestId 请求标识，为空时会自动生成

        @Url 异步回调地址
        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition,'url':url,'requestId':requestId}
        result=doPost(self.__textFilterConfig,self.__MarkdownReplaceUrl,request)
        return json.loads(result)

 
 
def doPost(textFilterConfig,url,postData):
    u=textFilterConfig.GetTextFilterHost()+url
    result = requests.post(u, postData)
    return  result.text